climada_petals/hazard/rf_glofas/rf_glofas.py

```diff
@@ -0,0 +1,166 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+User Interface for GloFAS River Flood Module
+"""
+
+import logging
+from contextlib import contextmanager
+from typing import Union
+
+import xarray as xr
+import pandas as pd
+from dask.distributed import Client
+
+from climada.util.constants import SYSTEM_DIR
+from climada.hazard import Hazard
+
+LOGGER = logging.getLogger(__name__)
+
+DEFAULT_DATA_DIR = SYSTEM_DIR / "river-flood-computation"
+
+
+@contextmanager
+def dask_client(n_workers, threads_per_worker, memory_limit, *args, **kwargs):
+    """Create a context with a ``dask.distributed.Client``.
+
+    This is a lightweight wrapper and intended to expose only the most important
+    parameters to end users.
+
+    Parameters
+    ----------
+    n_workers : int
+        Number of parallel processes to launch.
+    threads_per_worker : int
+        Compute threads launched by each worker.
+    memory_limit : str
+        Memory limit for each process. Example: 4 GB can be expressed as ``4000M`` or
+        ``4G``.
+    args, kwargs
+        Additional (keyword) arguments passed to the ``dask.distributed.Client``
+        constructor.
+
+    Example
+    -------
+    >>> with dask_client(n_workers=2, threads_per_worker=2, memory_limit="4G"):
+    ...     xr.open_dataset("data.nc", chunks="auto").median()
+    """
+    # Yield the client with the arguments, and close it afterwards
+    LOGGER.info("Creating dask.distributed.Client")
+    client = Client(
+        *args,
+        n_workers=n_workers,
+        threads_per_worker=threads_per_worker,
+        memory_limit=memory_limit,
+        **kwargs,
+    )
+    try:
+        yield client
+    finally:
+        LOGGER.info("Closing dask.distributed.Client")
+        client.close()
+
+
+def hazard_series_from_dataset(
+    data: xr.Dataset, intensity: str, event_dim: str
+) -> Union[pd.Series, Hazard]:
```

Comment:
I added tests for `hazard_series_from_dataset` based on your suggestion in aa0cdd2253f5f0d6e4a920d5cfc7bf96f2e26419. For me, your tests actually passed!

Reply:
Your tests only ever test the case where the `event_dim` variable is a datetime dimension. But the example given in your docstring uses `"number"` as the `event_dim` and in that case, I would not expect this dimension to be datetime-like.

Accordingly, this is what I (still) get when running my suggested test:
```Python
======================================================================
ERROR: test_hazard_series_from_dataset (climada_petals.hazard.rf_glofas.test.test_rf_glofas.TestHazardSeriesFromDataset)
Test the hazard_series_from_dataset function
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tovogt/code/climada_petals/climada_petals/hazard/rf_glofas/test/test_rf_glofas.py", line 118, in test_hazard_series_from_dataset
    haz_series = hazard_series_from_dataset(ds, "intensity", "number")
  File "/home/tovogt/code/climada_petals/climada_petals/hazard/rf_glofas/rf_glofas.py", line 160, in hazard_series_from_dataset
    hazards = [
  File "/home/tovogt/code/climada_petals/climada_petals/hazard/rf_glofas/rf_glofas.py", line 161, in <listcomp>
    create_hazard(data.sel(dict(zip(iter_dims, idx))))
  File "/home/tovogt/code/climada_petals/climada_petals/hazard/rf_glofas/rf_glofas.py", line 144, in create_hazard
    return Hazard.from_xarray_raster(
  File "/home/tovogt/code/climada_python/climada/hazard/base.py", line 843, in from_xarray_raster
    data[coords["event"]].dt.strftime("%Y-%m-%d").values.flatten().tolist(),
  File "/home/tovogt/.local/share/miniconda3/envs/climada_env_3.9/lib/python3.9/site-packages/xarray/core/utils.py", line 1108, in __get__
    return self._accessor(obj)  # type: ignore  # assume it is a valid accessor!
  File "/home/tovogt/.local/share/miniconda3/envs/climada_env_3.9/lib/python3.9/typing.py", line 687, in __call__
    result = self.__origin__(*args, **kwargs)
  File "/home/tovogt/.local/share/miniconda3/envs/climada_env_3.9/lib/python3.9/site-packages/xarray/core/accessor_dt.py", line 623, in __new__
    raise TypeError(
TypeError: '.dt' accessor only available for DataArray with datetime64 timedelta64 dtype or for arrays containing cftime datetime objects.

----------------------------------------------------------------------
```

Reply:
Indeed, I get the same error. It seems like https://github.com/CLIMADA-project/climada_python/pull/795 broke my code because it now requires either explicit event names or that the "event" dimension can be interpreted as DateTime, which was not a requirement before.

Reply:
https://github.com/CLIMADA-project/climada_python/pull/837 should have resolved this. I'll add an appropriate test case.